kongswap_adaptor/src/deposit/deposit_add_liquidity.rs

```diff
@@ -0,0 +1,800 @@
+
+use super::*;
+use crate::kong_types::{
+    AddLiquidityReply, AddPoolArgs, AddTokenArgs, ICReply, RemoveLiquidityAmountsArgs,
+    RemoveLiquidityAmountsReply, UpdateTokenArgs, UpdateTokenReply, UserBalanceLPReply,
+    UserBalancesArgs, UserBalancesReply,
+};
+use crate::KONG_BACKEND_CANISTER_ID;
+use crate::{
+    state::storage::ConfigState, validation::ValidatedTreasuryManagerInit, StableAuditTrail,
+    StableBalances, AUDIT_TRAIL_MEMORY_ID, BALANCES_MEMORY_ID,
+};
+use std::fmt::Debug;
+
+const E8: u64 = 100_000_000;
+const FEE_SNS: u64 = 10_500u64;
+const FEE_ICP: u64 = 9_500u64;
+
+use lazy_static::lazy_static;
+
+pub struct UnsafeSyncCell<T>(pub UnsafeCell<T>);
+
+impl<T> UnsafeSyncCell<T> {
+    pub(crate) fn new(value: T) -> Self {
+        Self(UnsafeCell::new(value))
+    }
+}
+
+// SAFETY: You must ensure synchronization yourself!
+unsafe impl<T: Send> Sync for UnsafeSyncCell<T> {}
+unsafe impl<T: Send> Send for UnsafeSyncCell<T> {}
+lazy_static! {
+    static ref SELF_CANISTER_ID: Principal =
+        Principal::from_text("jexlm-gaaaa-aaaar-qalmq-cai").unwrap();
+    static ref SNS_LEDGER: Principal = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    static ref ICP_LEDGER: Principal = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+    static ref SNS_ID: Principal = Principal::from_text("jg2ra-syaaa-aaaaq-aaewa-cai").unwrap();
+
+    static ref TOKEN_0: String = format!("IC.{}", *SNS_LEDGER);
+    static ref TOKEN_1: String = format!("IC.{}", *ICP_LEDGER);
+
+    static ref TOKEN_0_SYMBOL: String = "DAO".to_string();
+    static ref TOKEN_1_SYMBOL: String = "ICP".to_string();
+
+    // Create test assets and request first
+    static ref ASSET_0: Asset = Asset::Token {
+        ledger_canister_id: *SNS_LEDGER,
+        symbol: (*TOKEN_0_SYMBOL).to_string(),
+        ledger_fee_decimals: Nat::from(FEE_SNS),
+    };
+
+    static ref ASSET_1: Asset = Asset::Token {
+        ledger_canister_id: *ICP_LEDGER,
+        symbol: (*TOKEN_1_SYMBOL).to_string(),
+        ledger_fee_decimals: Nat::from(FEE_ICP),
+    };
+
+    static ref OWNER_ACCOUNT: sns_treasury_manager::Account = sns_treasury_manager::Account {
+        owner: Principal::from_text("2vxsx-fae").unwrap(),
+        subaccount: None,
+    };
+
+}
+
+lazy_static! {
+    // Add fields to control mock behavior
+    static ref CURRENT_STACK: UnsafeSyncCell<usize> = UnsafeSyncCell::new(0);
+    static ref CURRENT_INSTRUCTION: UnsafeSyncCell<usize> = UnsafeSyncCell::new(0);
+    static ref CALL_STACKS: UnsafeSyncCell<HashMap<usize, VecDeque<CallSpec>>> = UnsafeSyncCell::new(HashMap::<usize, VecDeque<CallSpec>>::new());
+    // source stack id, source instruction id, destination stack id, destination instruction id
+    static ref SWITCHING_CONTEXT: UnsafeSyncCell<Option<(usize, usize, usize, usize)>> = UnsafeSyncCell(UnsafeCell::new(None));
+}
+
+#[derive(Clone, Debug)]
+pub struct MockError {
```

Comment:
Maybe dumb question (b/c I don't have enough context on this project), but this seems like test code, but the part of the file structure it lives in is production code, and I don't see module definitions that would tell you it's just for tests.